iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/storable_function.inl

```diff
@@ -181,22 +167,13 @@ void storable_function<S, signature<ReturnType, Args...>>::storeFunctor(const Fu
         std::cerr << "storable_function: no memory to store functor\n";
         // this cannot happen in the static_storage case
     }
-
-    cxx::Ensures(!empty());
```

Comment:
You hit a nerve here. Potentially the allocator could use heap here and then we have to do a full blown error handling here despite the default use case is a compile time stack allocator.
This may needs a little refactoring.

Reply:
So I removed the template argument `StorageType` and used it as an internal type alias. Furthermore I added a `cxx::Ensures(ptr!=nullptr)` everywhere as you suggested.
Since the `StorageType` guarantees that the function can acquire memory at compile time we only run into trouble whenever this type alias is again a template argument where the user can use a different kind of allocator.